ciongke/cmd/hook/main.go

```diff
@@ -188,55 +190,53 @@ func (s *Server) handleEvent(e Event) {
 	}
 }
 
-func (s *Server) handlePullRequest(pr gogithub.PullRequestEvent) error {
-	if pr.Action == nil || pr.PullRequest == nil {
-		return fmt.Errorf("incomplete pull request event: %+v", pr)
-	}
-	switch *pr.Action {
+// handlePullRequest decides what to do with PullRequestEvents.
+func (s *Server) handlePullRequest(pr github.PullRequestEvent) error {
+	switch pr.Action {
 	case "opened", "reopened", "synchronize":
-		valid, err := s.validAuthor(*pr.PullRequest.User.Login)
+		valid, err := s.validAuthor(pr.PullRequest.User.Login)
 		if err != nil {
 			return fmt.Errorf("could not validate author: %s", err)
 		}
 		if valid {
 			if err := s.buildPR(pr.PullRequest); err != nil {
 				return fmt.Errorf("could not build PR: %s", err)
 			}
-		} else if *pr.Action == "opened" {
+		} else if pr.Action == "opened" {
 			// TODO: Comment asking them to join the team.
 		}
 	case "closed":
-		if err := s.deletePR(*pr.PullRequest.Number); err != nil {
+		if err := s.deletePR(pr.Number); err != nil {
 			return fmt.Errorf("could not delete old PR: %s", err)
 		}
 	}
 	return nil
 }
 
-// TODO: Cache this or use conditionals so that we don't get rate-limited.
+// validAuthor first checks if author is in the org, then the team.
 func (s *Server) validAuthor(author string) (bool, error) {
-	orgMember, _, err := s.GitHubClient.Organizations.IsMember(s.Org, author)
+	orgMember, err := s.GitHubClient.IsMember(s.Org, author)
 	if err != nil {
 		return false, err
 	}
 	if orgMember {
 		return true, nil
 	}
-	teamMember, _, err := s.GitHubClient.Organizations.IsTeamMember(s.TeamID, author)
+	teamMember, err := s.GitHubClient.IsTeamMember(s.TeamID, author)
 	if err != nil {
 		return false, err
 	}
 	return teamMember, nil
 }
 
-// buildPR starts a job to build the PR.
-func (s *Server) buildPR(pr *gogithub.PullRequest) error {
+// buildPR deletes any jobs building the PR and then starts a new one.
+func (s *Server) buildPR(pr github.PullRequest) error {
 	job := &batch.Job{
 		ObjectMeta: api.ObjectMeta{
-			Name:      "pr-" + strconv.Itoa(*pr.Number) + "-" + *pr.Head.SHA,
+			Name:      fmt.Sprintf("pr-%d-%s", pr.Number, pr.Head.SHA),
```

Comment:
Should include the repo owner and name in the name/labels, to make multi-repo testing more doable.